### DISCUSSION TITLE
Cannot build Windows Client

### DISCUSSION THREAD

**Discussion by fundave3** [rustdesk/rustdesk#9693] [General]
Hi i am trying to build the windows client.
Ive installed Flutter on the Windows System
Visual Studio Code and the vcpkg 

After cloning the github repo i am trying to build the windows client with  flutter build windows.

I am facing some issues 

lib/models/platform_model.dart(2,8): error GFAA2A68C: Error when reading 'lib/generated_bridge.dart': The system cannot find the file specified. [C:\Users\Administrator\rustdesk\flutter\build\windows\x64\flutter\flutter_assemble.vcxproj]
lib/models/model.dart(41,8): error GFAA2A68C: Error when reading 'lib/generated_bridge.dart': The system cannot find the file specified. [C:\Users\Administrator\rustdesk\flutter\build\windows\x64\flutter\flutter_assemble.vcxproj]
lib/models/native_model.dart(16,8): error GFAA2A68C: Error when reading 'lib/generated_bridge.dart': The system cannot find the file specified. [C:\Users\Administrator\rustdesk\flutter\build\windows\x64\flutter\flutter_assemble.vcxproj]
lib/models/platform_model.dart(8,1): error G5FE39F1E: Type 'RustdeskImpl' not found. [C:\Users\Administrator\rustdesk\flutter\build\windows\x64\flutter\flutter_assemble.vcxproj]
lib/models/native_model.dart(35,8): error G5FE39F1E: Type 'RustdeskImpl' not found. [C:\Users\Administrator\rustdesk\flutter\build\windows\x64\flutter\flutter_assemble.vcxproj]
lib/models/native_model.dart(44,3): error G5FE39F1E: Type 'RustdeskImpl' not found. [C:\Users\Administrator\rustdesk\flutter\build\windows\x64\flutter\flutter_assemble.vcxproj]
lib/models/native_model.dart(239,26): error G5FE39F1E: Type 'RustdeskImpl' not found. [C:\Users\Administrator\rustdesk\flutter\build\windows\x64\flutter\flutter_assemble.vcxproj]
lib/common.dart(1922,48): error GC2F972A8: The argument type 'String?' can't be assigned to the parameter type 'String' because 'String?' is nullable and 'String' isn't. [C:\Users\Administrator\rustdesk\flutter\build\windows\x64\flutter\flutter_assemble.vcxproj]
lib/models/model.dart(2518,23): error G1D2B1123: 'EventToUI' isn't a type. [C:\Users\Administrator\rustdesk\flutter\build\windows\x64\flutter\flutter_assemble.vcxproj]
lib/models/model.dart(2571,24): error GBAFEC12A: 'EventToUI_Event' isn't a type. [C:\Users\Administrator\rustdesk\flutter\build\windows\x64\flutter\flutter_assemble.vcxproj]
lib/models/model.dart(2587,31): error GACE86540: 'EventToUI_Rgba' isn't a type. [C:\Users\Administrator\rustdesk\flutter\build\windows\x64\flutter\flutter_assemble.vcxproj]
lib/models/model.dart(2602,31): error G72B918CB: 'EventToUI_Texture' isn't a type. [C:\Users\Administrator\rustdesk\flutter\build\windows\x64\flutter\flutter_assemble.vcxproj]
lib/desktop/pages/desktop_setting_page.dart(1513,33): error GC2F972A8: The argument type 'Future<Null> Function(String)?' can't be assigned to the parameter type 'dynamic Function(dynamic)?'. [C:\Users\Administrator\rustdesk\flutter\build\windows\x64\flutter\flutter_assemble.vcxproj]
lib/desktop/pages/desktop_setting_page.dart(1518,33): error GC2F972A8: The argument type 'Future<Null> Function(String)?' can't be assigned to the parameter type 'dynamic Function(dynamic)?'. [C:\Users\Administrator\rustdesk\flutter\build\windows\x64\flutter\flutter_assemble.vcxproj]
lib/desktop/pages/desktop_setting_page.dart(1536,33): error GC2F972A8: The argument type 'Future<Null> Function(String)?' can't be assigned to the parameter type 'dynamic Function(dynamic)?'. [C:\Users\Administrator\rustdesk\flutter\build\windows\x64\flutter\flutter_assemble.vcxproj]
lib/desktop/pages/desktop_setting_page.dart(1541,33): error GC2F972A8: The argument type 'Future<Null> Function(String)?' can't be assigned to the parameter type 'dynamic Function(dynamic)?'. [C:\Users\Administrator\rustdesk\flutter\build\windows\x64\flutter\flutter_assemble.vcxproj]
lib/desktop/pages/desktop_setting_page.dart(1559,33): error GC2F972A8: The argument type 'Future<Null> Function(String)?' can't be assigned to the parameter type 'dynamic Function(dynamic)?'. [C:\Users\Administrator\rustdesk\flutter\build\windows\x64\flutter\flutter_assemble.vcxproj]
lib/desktop/pages/desktop_setting_page.dart(1564,33): error GC2F972A8: The argument type 'Future<Null> Function(String)?' can't be assigned to the parameter type 'dynamic Function(dynamic)?'. [C:\Users\Administrator\rustdesk\flutter\build\windows\x64\flutter\flutter_assemble.vcxproj]
lib/desktop/pages/desktop_setting_page.dart(1569,33): error GC2F972A8: The argument type 'Future<Null> Function(String)?' can't be assigned to the parameter type 'dynamic Function(dynamic)?'. [C:\Users\Administrator\rustdesk\flutter\build\windows\x64\flutter\flutter_assemble.vcxproj]
lib/desktop/pages/desktop_setting_page.dart(1574,33): error GC2F972A8: The argument type 'Future<Null> Function(String)?' can't be assigned to the parameter type 'dynamic Function(dynamic)?'. [C:\Users\Administrator\rustdesk\flutter\build\windows\x64\flutter\flutter_assemble.vcxproj]
lib/desktop/pages/desktop_setting_page.dart(1602,35): error GC2F972A8: The argument type 'Future<Null> Function(String)?' can't be assigned to the parameter type 'dynamic Function(dynamic)?'. [C:\Users\Administrator\rustdesk\flutter\build\windows\x64\flutter\flutter_assemble.vcxproj]
lib/desktop/pages/desktop_setting_page.dart(1609,35): error GC2F972A8: The argument type 'Future<Null> Function(String)?' can't be assigned to the parameter type 'dynamic Function(dynamic)?'. [C:\Users\Administrator\rustdesk\flutter\build\windows\x64\flutter\flutter_assemble.vcxproj]
lib/desktop/pages/desktop_setting_page.dart(1619,33): error GC2F972A8: The argument type 'Future<Null> Function(String)?' can't be assigned to the parameter type 'dynamic Function(dynamic)?'. [C:\Users\Administrator\rustdesk\flutter\build\windows\x64\flutter\flutter_assemble.vcxproj]
lib/desktop/pages/desktop_setting_page.dart(1624,33): error GC2F972A8: The argument type 'Future<Null> Function(String)?' can't be assigned to the parameter type 'dynamic Function(dynamic)?'. [C:\Users\Administrator\rustdesk\flutter\build\windows\x64\flutter\flutter_assemble.vcxproj]
lib/desktop/pages/desktop_setting_page.dart(1629,33): error GC2F972A8: The argument type 'Future<Null> Function(String)?' can't be assigned to the parameter type 'dynamic Function(dynamic)?'. [C:\Users\Administrator\rustdesk\flutter\build\windows\x64\flutter\flutter_assemble.vcxproj]
lib/desktop/pages/desktop_setting_page.dart(1634,33): error GC2F972A8: The argument type 'Future<Null> Function(String)?' can't be assigned to the parameter type 'dynamic Function(dynamic)?'. [C:\Users\Administrator\rustdesk\flutter\build\windows\x64\flutter\flutter_assemble.vcxproj]
lib/common/widgets/dialog.dart(1303,21): error G44692867: A value of type 'String?' can't be assigned to a variable of type 'String' because 'String?' is nullable and 'String' isn't. [C:\Users\Administrator\rustdesk\flutter\build\windows\x64\flutter\flutter_assemble.vcxproj]
lib/models/native_model.dart(35,8): error G4FA4B0A1: 'RustdeskImpl' isn't a type. [C:\Users\Administrator\rustdesk\flutter\build\windows\x64\flutter\flutter_assemble.vcxproj]
lib/models/native_model.dart(142,18): error GE5CFE876: The method 'RustdeskImpl' isn't defined for the class 'PlatformFFI'. [C:\Users\Administrator\rustdesk\flutter\build\windows\x64\flutter\flutter_assemble.vcxproj]
lib/models/native_model.dart(239,26): error G4FA4B0A1: 'RustdeskImpl' isn't a type. [C:\Users\Administrator\rustdesk\flutter\build\windows\x64\flutter\flutter_assemble.vcxproj]
lib/common/widgets/toolbar.dart(456,39): error G44692867: A value of type 'bool?' can't be assigned to a variable of type 'bool' because 'bool?' is nullable and 'bool' isn't. [C:\Users\Administrator\rustdesk\flutter\build\windows\x64\flutter\flutter_assemble.vcxproj]
C:\Program Files\Microsoft Visual Studio\2022\Community\MSBuild\Microsoft\VC\v170\Microsoft.CppCommon.targets(254,5): error MSB8066: Custom build for 'C:\Users\Administrator\rustdesk\flutter\build\windows\x64\CMakeFiles\054a4a9c136604ac524498fc6f6b5c72\flutter_windows.dll.rule;C:\Users\Administrator\rustdesk\flutter\build\windows\x64\CMakeFiles\adbb6e2f9ab3f882b63155d4400e406d\flutter_assemble.rule' exited with code 1. [C:\Users\Administrator\rustdesk\flutter\build\windows\x64\flutter\flutter_assemble.vcxproj]

Any Idea whats wrong ?

**Comment by rustdesk** [comment]
https://github.com/rustdesk/rustdesk/discussions/763, go to [discord](https://discord.gg/nDceKgxnkV) please, please search there first, many guys have asked similar question because you are not the most special though you always think you are. Usually nobody will help you with build or self-hosting here.
